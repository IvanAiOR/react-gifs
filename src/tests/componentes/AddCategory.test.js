import React from 'react';
const { shallow } = require("enzyme");
const { AddCategory } = require("../../componentes/AddCategory")
import '@testing-library/jest-dom';

describe('Pruebas en <AddCategory/>', () => {
    const setCategory = jest.fn();
    let wrapper = shallow(<AddCategory setCategorias={setCategory} />);

    beforeEach(()=>{
        jest.clearAllMocks();//esta funcion limpia todos los valores que hayamos usado en una prueba anterior en el mismo documento
        wrapper= shallow(<AddCategory setCategorias={setCategory} />);
    });
    
    test('debe mostrarse correctamente', () => {
            expect(wrapper).toMatchSnapshot();
    });
    
    test('debe mostrar un texto en la caja', () => {
        const input= wrapper.find('input');
        const value='hola mundo';

        input.simulate('change',{target:{value}})// no es necesario escribir el value como propiedad y valor porque automaticamente react lo interpreta
        
        expect(wrapper.find('p').text().trim()).toBe(value);
    });
    
    test('debe señalar el submit limpio', () => {
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        expect(setCategory).not.toHaveBeenCalled(); //sile quitas el not, consulta por las veces que se llamo el submit, pero por el "preventDefault" lo tomará como error
    });

    test('debe probar si el input se guarda y ademas envía el submit una vez', () => {
        
        const value='Hola mundo';

        wrapper.find('input').simulate('change',{target:{value}});
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        
        expect(setCategory).toHaveBeenCalled();//espera que se llame 
        expect(setCategory).toBeCalledWith(expect.any(Function));//espera que se llame 

    });
    
    
})
