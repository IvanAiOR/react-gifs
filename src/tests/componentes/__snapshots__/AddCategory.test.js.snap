// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory/> debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <h3>
    Ingresa una categoria
  </h3>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <button
    type="submit"
  >
    Agregar
  </button>
</form>
`;
